Applications/Viqui/vqCore.cxx

```diff
@@ -3387,6 +3393,37 @@ void vqCore::generateReport(QString path, bool generateVideo)
   this->exportKml(path);
 }
 
+//-----------------------------------------------------------------------------
+void vqCore::exportSvm(QString path)
+{
+  QFile file(path);
+  if (!file.open(QIODevice::WriteOnly |
+                 QIODevice::Truncate))
+    {
+    qDebug() << file.errorString();
+    QMessageBox::warning(0, QString(), "Unable to write SVM file.");
+    return;
+    }
+
+  if (this->SavedQueryPlan.isSimilarityQuery())
+    {
+    auto& query = *this->SavedQueryPlan.constSimilarityQuery();
+
+    const int iqrModelSize = static_cast<int>(query.IqrModel.size());
+    if (iqrModelSize)
+      {
+      const char* rawModel =
+        reinterpret_cast<const char*>(&query.IqrModel[0]);
+      const QByteArray encodedModel =
+        QByteArray::fromRawData(rawModel, iqrModelSize).toBase64();
+
+      file.write(encodedModel);
+      }
+    }
+
+  file.close();
```

Comment:
Pedantic: this is superfluous; the QFile is closed on destruction.

Reply:
True, but... I'm of the mindset that explicit is better than implicit, and this isn't completely obvious to someone who's not familiar with `QFile`.

It's fine, and I'll take it out if you want, but it's not what I'd recommend to other developers.

Reply:
It... would be more consistent, I think. Out of a non-trivial number of uses of file I/O (including std::fstream as well as QIODevice), there are 3.5 instances (not counting this one) where `close()` is called:

```
Applications/VpView/vpGraphModelWidget.cxx:1135:    file.close();
Applications/VpView/vpView.cxx:3043:  file.close();
Tools/KmlWriter/KmlWriter.cxx:224:    out.close();
Tools/KmlWriter/KmlWriter.cxx:230:    out.close();
```

The last two are the same object, hence counting them as "1.5"; they aren't really "separate" instances. The second actually looks important; there is possibly an external process executed between the call to `close()` and the `QFile` going out of scope, that I would guess may use the file, so that one is justified. The overall pattern appears to be to not explicitly call `close()` unless there is a reason for doing so.